pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +879,56 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+				Name:      cdName,
+				Namespace: pod.Namespace,
+			}})
+		}
+	}
+	return retval
+}
+
+// Calculates the restart count for both hive and installer container for a cd and sets it to metricClusterDeploymentInstallRetries
+func (r *ReconcileClusterDeployment) updateInstallRetriesTotalMetric(clusterDeploymentName string, clusterDeploymentNamespace string, cdLog log.FieldLogger) {
+	labels := map[string]string{"cluster_deployment": clusterDeploymentName, "namespace": clusterDeploymentNamespace}
+	installerPodLabels := map[string]string{install.ClusterDeploymentNameLabel: clusterDeploymentName, install.InstallJobLabel: "true"}
+	_, err := metricClusterDeploymentInstallRetries.GetMetricWith(labels)
```

Comment:
Don't think we need this anymore now that we're not going to try to read a counters current value. Can drop the if/else below as well on the err check.